File: src/main/ng6-client/src/app/models/volume.ts
import {Titul} from 'src/app/models/titul';
import { PeriodicitaSvazku } from 'src/app/models/periodicita-svazku';
import { WeekDay } from '@angular/common';

export class Volume {
  id: string;
  id_titul: string;
  titul: Titul;
  mutace: string;
  znak_oznaceni_vydani: string;
  carovy_kod: string;
  signatura: string;
  vlastnik: string;
  periodicita: PeriodicitaSvazku[];
  prvni_cislo: number;
  posledni_cislo: number;
  poznamka: string;
  datum_od: Date = new Date();
  datum_do: Date = new Date();

  constructor() {
    this.periodicita = [];
      const wds: string[] = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];
      wds.forEach(wd => {
        const ps = new PeriodicitaSvazku();
        ps.den = wd;
        this.periodicita.push(ps);
      });
  }
}

File: src/main/java/cz/incad/nkp/inprove/entities/volume/Volume.java

package cz.incad.nkp.inprove.entities.volume;

import cz.incad.nkp.inprove.entities.DomainObject;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.solr.core.mapping.Indexed;
import org.springframework.data.solr.core.mapping.SolrDocument;

import java.util.Date;
import java.util.List;

import static cz.incad.nkp.inprove.entities.volume.Volume.CORE_NAME;

@Getter
@Setter
@SolrDocument(collection = CORE_NAME)
public class Volume extends DomainObject {

    public static final String CORE_NAME = "svazek";

    @Indexed(name = "id_titul", type = "string")
    private String id_titul;

    @Indexed(name = "signatura", type = "string")
    private String signatura;

    @Indexed(name = "mutace", type = "string")
    private String mutace;

    @Indexed(name = "periodicita", type = "string")
    private List<String> periodicita;

    @Indexed(name = "poznamka", type = "string")
    private String poznamka;

    @Indexed(name = "vlastnik", type = "string")
    private String vlastnik;

    @Indexed(name = "datum_od", type = "rdate")
    private Date datum_od;

    @Indexed(name = "datum_do", type = "rdate")
    private Date datum_do;

    @Indexed(name = "prvni_cislo", type = "string")
    private String prvni_cislo;

    @Indexed(name = "posledni_cislo", type = "string")
    private String posledni_cislo;

    @Indexed(name = "znak_oznaceni_vydani", type = "string")
    private String znak_oznaceni_vydani;

    @Indexed(name = "carovy_kod", type = "string")
    private String carovy_kod;

    @Indexed(name = "pocet_stran", type = "plong")
    private Long pocet_stran;

    @Indexed(name = "show_attachments_at_the_end", type = "boolean")
    private Boolean show_attachments_at_the_end;
}
